Domain/EventMigrator.cs

```diff
@@ -59,5 +61,19 @@ public Rename(long sequenceNumber, string newName)
                 NewName = newName;
             }
         }
+
+        public static async Task SaveWithRenames<TAggregate>(IEventSourcedRepository<TAggregate> repository, TAggregate aggregate, IEnumerable<Rename> renames)
+            where TAggregate : EventSourcedAggregate<TAggregate>
+        {
+            if (repository == null)
+            {
+                throw new ArgumentNullException("repository");
+            }
+            if (!(repository is IMigratableEventSourcedRepository<TAggregate>))
+            {
+                throw new RepositoryMustSupportMigrationsException(repository.GetType());
```

Comment:
Rather than a runtime surprise, I think this could be enforced at compile time using a generic constraint, e.g. 

```
public static async Task SaveWithRenames<TRepository, TAggregate>(TRespository ...)
     where TRepository : IEventSourcedRepository<TAggregate> ...
```